ghcide/src/Development/IDE/Core/Compile.hs

```diff
@@ -216,23 +310,44 @@ captureSplices env k = do
             pure $ f aw'
 
 
-tcRnModule :: HscEnv -> [Linkable] -> ParsedModule -> IO TcModuleResult
+tcRnModule :: HscEnv -> ModuleEnv UTCTime -> ParsedModule -> IO TcModuleResult
 tcRnModule hsc_env keep_lbls pmod = do
   let ms = pm_mod_summary pmod
       hsc_env_tmp = hscSetFlags (ms_hspp_opts ms) hsc_env
+      hpt = hsc_HPT hsc_env
 
-  unload hsc_env_tmp keep_lbls
+  unload hsc_env_tmp $ map (\(mod, time) -> LM time mod []) $ moduleEnvToList keep_lbls
```

Comment:
This seems to be unloading too much, as `keep_lbls` omits the transitive dependencies. Why is this ok?